File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    
    bool binarySearch(vector<vector<int>> &A, int target, int start, int end){
        if(start <= end){
            int mid = (start + end);
            int i = mid/A[0].size();
            int j = mid%A[0].size();
            
            if(target == A[i][j]) return true;
            if(target > A[i][j]) return binarySearch(A, target, mid + 1, end);
            else return binarySearch(A, target, start, mid - 1);
        }
        return false;
    }
    
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        return binarySearch(matrix, target, 0, matrix.size()*matrix[0].size() - 1);
    }
};

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        i, j = 0, len(matrix[0]) - 1
        
        while (i >=0 and i < len(matrix)) and (j >= 0 and j < len(matrix[0])):
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] > target:
                j -= 1
            else:
                i += 1
                
        return False